src/controllers/hid/hidcontroller.cpp

```diff
@@ -101,18 +100,36 @@ int HidController::open() {
     // Set hid controller to non-blocking
     if (hid_set_nonblocking(m_pHidDevice, 1) != 0) {
         qCWarning(m_logBase) << "Unable to set HID device " << getName() << " to non-blocking";
+        hid_close(m_pHidDevice);
         return -1;
     }
 
-    // This isn't strictly necessary but is good practice.
-    for (int i = 0; i < kNumBuffers; i++) {
-        memset(m_pPollData[i], 0, kBufferSize);
-    }
-    m_lastPollSize = 0;
-
     setOpen(true);
+
+    m_pHidIoThread = std::make_unique<HidIoThread>(m_pHidDevice, std::move(m_deviceInfo));
```

Comment:
std::move is meant to only be used when the thing being moved will be permanently be owned by the receiving function and not used again in the calling context.  Because this is a member variable, this seems unlikely.  This is not a bug, but it is unlikely that the std::move is doing what you think it's doing.  If the deviceinfo is not needed after this move, then it shouldn't be a member.  if it is used, then drop std::move and just make a copy (it is a cheap copy only done once.)

some background reference: https://stackoverflow.com/questions/42795683/using-an-object-after-stdmove-doesnt-result-in-a-compilation-error